services/src/main/scala/cromwell/pipeline/service/ProjectService.scala

```diff
@@ -39,32 +45,25 @@ class ProjectService(projectRepository: ProjectRepository, projectVersioning: Pr
     }
   }
 
-  def deactivateProjectById(projectId: ProjectId, userId: UserId): Future[Option[Project]] = {
-    val result = getProjectById(projectId)
-    result.flatMap {
-      case Some(project) if project.ownerId != userId => Future.failed(new ProjectAccessDeniedException)
-      case Some(_) =>
-        for {
-          _ <- projectRepository.deactivateProjectById(projectId)
-          getProject <- getProjectById(projectId)
-        } yield getProject
-      case None => Future.failed(new ProjectNotFoundException)
+  def deactivateProjectById(projectId: ProjectId, userId: UserId): Future[Project] =
+    getProjectById(projectId, userId).flatMap { project =>
+      if (!project.active) {
+        Future.failed(ProjectIllegalStateException(s"Cannot deactivate non-active project"))
```

Comment:
why we should fail if project already deactivated? just do nothing and return project?

Reply:
IDK, I thought deactivation of a deactivated project should not be allowed. Anyway, I decided to return as it was and not return an error in this case.